### PR TITLE
Update PlutusV3 ScriptContext

### PR DESCRIPTION
**Pull request by zliu41**
The following changes are made:
- `GovernanceActionId` is made optional
- `Proposing` now takes an Integer argument
- `ParameterChange` now takes a `ChangedParameters` argument.
- Removed `Anchor`

### CODE REVIEW

**Review by WhatisRT — APPROVED** [review]
LGTM

**Comment by michaelpj** [comment]
Are we going to do the anchor stuff too?

**Comment by zliu41** [comment]
If this looks good I'll do the anchor stuff separately

File: plutus-benchmark/script-contexts/test/checkScriptContextEqualityTerm-20.budget.golden [hunk 1]
```diff
@@ -1,2 +1,2 @@
-({cpu: 603616251
-| mem: 2466446})
\ No newline at end of file
+({cpu: 603524251
```
Comment (zliu41) [inline plutus-benchmark/script-contexts/test/checkScriptContextEqualityTerm-20.budget.golden:1]: @michaelpj This suggests a potential improvements to the compiler - A data constructor application should be considered pure if all arguments are pure.

**Review by zliu41 — COMMENTED** [review]
Or if this is still queued for CI when I start working on the Anchor stuff then I'll just do it here.

File: plutus-benchmark/script-contexts/test/checkScriptContextEqualityTerm-20.budget.golden (hunk 1 above)
Comment (michaelpj) [reply plutus-benchmark/script-contexts/test/checkScriptContextEqualityTerm-20.budget.golden:1]: Yeah, I commented on this in the other PR too.

**Comment by zliu41** [comment]
Removed anchor everywhere: [9fefc90](https://github.com/input-output-hk/plutus/pull/5481/commits/9fefc90e29b30132fecc8596393393b150d3ddcb)

**Pull request merged**